nc/source/ecs/EcsApiImpl.cpp

```diff
@@ -16,7 +16,7 @@ namespace nc
 
     EntityHandle CreateEntity(EntityInfo info)
     {
-        IF_THROW(info.scale == Vector3::Zero(), "CreateEntity - scale cannot be Vector3::Zero");
+        IF_THROW(info.scale.x == 0.0f || info.scale.y == 0.0f || info.scale.z == 0.0f, "CreateEntity - Invalid scale(elements cannot be 0)");
```

Comment:
I still want to find a way to extract this elsewhere

Reply:
Done, there is a function to do the validation, and it is called only in Transform/Collider.